web/src/pages.rs

```diff
@@ -146,3 +151,77 @@ pub async fn my_articles(
     })
     .into_response())
 }
+
+#[axum::debug_handler(state = AppState)]
+pub async fn my_categories(
+    t: Template,
+    State(state): State<AppState>,
+    user: User,
+) -> WebResult<Response> {
+    info!("Fetching categories for user_id: {}", user.user_id);
+
+    let categories = sqlx::query!(
+        "SELECT DISTINCT c.* 
+     FROM category c
+     JOIN categorymarkdown cm ON c.category_id = cm.category_id
+     JOIN markdown m ON cm.markdown_id = m.markdown_id
+     JOIN pagesnapshot ps ON m.page_snapshot_id = ps.page_snapshot_id
+     JOIN pages p ON ps.page_id = p.page_id
+     WHERE p.user_id = $1",
+        user.user_id
+    )
+    .fetch_all(&state.db)
+    .await?;
+
+    Ok(t.render(maud::html! {
+        h1 { "My Categories" }
+        ul {
+            @for category in categories {
+                @let category_name = category.category.unwrap_or("No category".to_string());
```

Comment:
Hmm why is `category.category` an Option?

If we created a `Category` row in the table wouldn't we expect it to have a Category? Is that column null-able and if so should it be lol?

Reply:
because we have column name in `category` table  that is named `category`

<img width="455" alt="Screenshot 2024-07-09 at 11 08 39 AM" src="https://github.com/coreyja/knowledge/assets/55955558/3d832e71-864c-4441-8c99-cf5977bf7b2d">

Reply:
Ahh the answer to my question was that the column is marked as `is_nullable`, and thats why we get an option back!

We might eventually want to make that non-nullable, cause if there is a category row, I think there should be something in the category column lol
But that doesn't have to be a thing for this PR!